### PR TITLE
[#4483] add error retry to TestFetchFileUtils 

### PR DESCRIPTION
**Pull request by Naresh-kumar-Thodupunoori**
# feat: Implement Retry Mechanism in `TestFetchFileUtils` for Network-Related Issues

## Description
This pull request addresses the issue of inconsistent test results in the `TestFetchFileUtils` class due to network-related failures, specifically `SocketTimeoutExceptions`, by implementing a retry mechanism in the `testDownloadFromHTTP()` method.

## What changes were proposed in this pull request?
This PR introduces the following changes:

- **Retry Mechanism**: Added a configurable number of retry attempts to the `testDownloadFromHTTP()` method.
- **Exponential Backoff**: Implemented a delay between retry attempts using an exponential backoff strategy to reduce the frequency of retry attempts over time.
- **Asynchronous Handling**: Integrated `await()` for handling asynchronous retry logic, improving the test's robustness against transient network issues.
- **Enhanced Error Handling**: Improved error reporting for failed download attempts, providing more detailed feedback when retries are exhausted.

## Why are the changes needed?
These changes are necessary because:

- The existing implementation of `testDownloadFromHTTP()` was prone to failures caused by `SocketTimeoutExceptions`, leading to inconsistent test outcomes, particularly in environments with unreliable network connections.
- By introducing a retry mechanism, the test suite becomes more resilient to network instability, ensuring more consistent and reliable test results.
- The retry mechanism also ensures that the test will fail appropriately if persistent network issues occur, maintaining the integrity of the test suite.

**Fix**: #4483

## Does this PR introduce any user-facing change?
No, this PR does not introduce any user-facing changes. The updates are confined to the test suite and do not impact any user-facing APIs, functionality, or configuration properties.

## How was this patch tested?
The patch was tested using the following methods:

- Updated the `testDownloadFromHTTP()` test to include the retry logic and verified consistent pass results under stable network conditions.
- Conducted manual testing by deliberately disrupting network connectivity to ensure the retry mechanism triggers as expected.
- Executed the test over 100 iterations to confirm that the retry mechanism does not introduce any new flakiness or instability.
- Verified that the test fails appropriately after all retry attempts are exhausted.
- Ensured that the overall test execution time remains within acceptable limits under normal network conditions.

### CODE REVIEW

**Comment by Naresh-kumar-Thodupunoori** [comment]
@FANNG1 can you review my PR

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java [hunk 1]
```diff
@@ -43,9 +46,29 @@ public void testLinkLocalFile() throws Exception {
...
+
+    while (!success && attempts < MAX_RETRIES) {
+      try {
+        FetchFileUtils.fetchFileFromUri(fileUrl, destFile, 10, conf);
+        success = true;
+      } catch (IOException e) {
+        attempts++;
+        if (attempts < MAX_RETRIES) {
+          System.out.println(
+              "Attempt " + attempts + " failed. Retrying in " + RETRY_DELAY_MS + "ms.");
```
Comment (jerryshao) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java:63]: We don't allow to use `println`, please use log4j or remove this line.

**Comment by Naresh-kumar-Thodupunoori** [comment]
Ok, and would like me to change anything else!?

**Comment by Naresh-kumar-Thodupunoori** [comment]
@jerryshao Did the required changes.! If need to change anything else ping me!

**Comment by Naresh-kumar-Thodupunoori** [comment]
@jerryshao It is building fine in my local
image

**Comment by Naresh-kumar-Thodupunoori** [comment]
@jerryshao could help me with this. I am unable to fetch the error in the build using gradle as it is working fine in my local.

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java [hunk 2]
```diff
@@ -20,15 +20,20 @@
...
 import java.io.File;
+import java.io.IOException;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.log4j.Logger;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class TestFetchFileUtils {
 
+  private static final Logger logger = Logger.getLogger(TestFetchFileUtils.class);
```
Comment (FANNG1) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java:31]: please use slf4j
```java
Logger LOG = LoggerFactory.getLogger(xxx.class);
```

**Comment by FANNG1** [comment]
LGTM except one comment

File: catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java [hunk 3]
```diff
@@ -22,13 +22,14 @@
...
-import org.apache.log4j.Logger;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class TestFetchFileUtils {
 
-  private static final Logger logger = Logger.getLogger(TestFetchFileUtils.class);
+  Logger LOG = LoggerFactory.getLogger(TestFetchFileUtils.class);
```
Comment (FANNG1) [inline catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java:32]: please keep `private static final `

Comment (Naresh-kumar-Thodupunoori) [reply catalogs/catalog-hive/src/test/java/org/apache/gravitino/catalog/hive/TestFetchFileUtils.java]: did the needful!

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**